extensions/wikia/FeedsAndPosts/JustUpdatedAPIProxy.class.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace Wikia\FeedsAndPosts;
+
+class JustUpdatedAPIProxy {
+
+	const LIMIT = 4;
+	const MINOR_CHANGE_THRESHOLD = 100;
+
+	private function requestAPI( $params ) {
+		$api = new \ApiMain( new \FauxRequest( $params ) );
+		$api->execute();
+
+		return $api->GetResultData();
+	}
+
+	private function filterOutSmallChanges( $articles ) {
+		$articlesWithMinorChange = [];
+		$articlesWithMajorChange = [];
+
+		foreach ( $articles as &$article ) {
+			if ( abs( $article['newlen'] - $article['oldlen'] ) >= self::MINOR_CHANGE_THRESHOLD ) {
+				$articlesWithMajorChange[] = $article;
+			} else {
+				$articlesWithMinorChange[] = $article;
+			}
+		}
+
+		if ( count( $articlesWithMajorChange ) >= 4 ) {
+			return array_slice( $articlesWithMajorChange, 0, self::LIMIT );
+		}
+
+		return array_slice( array_merge( $articlesWithMajorChange, $articlesWithMinorChange ), 0,
+			self::LIMIT );
+	}
+
+	private function getRecentChanges() {
+		global $wgContentNamespaces;
+
+		$response = $this->requestAPI( [
+			'action' => 'query',
+			'list' => 'recentchanges',
+			'rcprop' => 'title|sizes',
+			'rcshow' => '!bot|!minor|!redirect',
+			'rcnamespace' => implode( '|', $wgContentNamespaces ),
+			// load 20 times more items so we'll be able to filter out articles with minor changes
+			'rclimit' => self::LIMIT * 20,
```

Comment:
100*20? why so much?

Reply:
I thought increasing limit to 100 would be enough

Reply:
it's `4*20`

Reply:
ok, I missread something